src/main/java/org/apache/mesos/offer/MesosResourcePool.java

```diff
@@ -89,38 +92,23 @@ public Offer getOffer() {
         return Optional.empty();
     }
 
-    /**
-     * Consumes and returns a {@link MesosResource} which meets the provided
-     * {@link DynamicPortRequirement}, or does nothing and returns an empty {@link Optional} if no
-     * available resources meet the requirement.
-     */
-    public Optional<MesosResource> consume(DynamicPortRequirement dynamicPortRequirement) {
-        Value availableValue = unreservedMergedPool.get(dynamicPortRequirement.getName());
+    public Optional<MesosResource> consumeDynamicPort(ResourceRequirement resReq){
+        Value availableValue = unreservedMergedPool.get("ports");
 
         if (availableValue == null) {
             return Optional.empty();
         }
 
-        // Choose first available port
         if (availableValue.getRanges().getRangeCount() > 0) {
             Value.Range range = availableValue.getRanges().getRange(0);
-            Resource resource = ResourceUtils.getUnreservedResource(
-                    dynamicPortRequirement.getName(),
-                    Value.newBuilder()
-                        .setType(Value.Type.RANGES)
-                        .setRanges(Value.Ranges.newBuilder()
-                                .addRange(Value.Range.newBuilder()
-                                        .setBegin(range.getBegin())
-                                        // Use getBegin again, since we just want the one port.
-                                        .setEnd(range.getBegin())))
-                        .build());
-
-            return consumeUnreservedMerged(new ResourceRequirement(resource));
-        }
+            resReq.addPort((int) range.getBegin());
```

Comment:
This shouldn't be part of the ResourceRequirement.  It should be part of the FulfilledRequirement.